packages/server/lib/controllers/appStoreAuth.controller.ts

```diff
@@ -0,0 +1,217 @@
+import type { Request, Response, NextFunction } from 'express';
+import type { LogLevel } from '@nangohq/shared';
+import {
+    getAccount,
+    getEnvironmentId,
+    createActivityLog,
+    errorManager,
+    analytics,
+    AnalyticsTypes,
+    connectionCreated as connectionCreatedHook,
+    createActivityLogMessage,
+    updateSuccess as updateSuccessActivityLog,
+    AuthCredentials,
+    updateProvider as updateProviderActivityLog,
+    configService,
+    connectionService,
+    createActivityLogMessageAndEnd,
+    AuthModes,
+    hmacService,
+    ErrorSourceEnum,
+    LogActionEnum
+} from '@nangohq/shared';
+
+class AppStoreAuthController {
+    async auth(req: Request, res: Response, next: NextFunction) {
+        const accountId = getAccount(res);
+        const environmentId = getEnvironmentId(res);
+        const { providerConfigKey } = req.params;
+        const connectionId = req.query['connection_id'] as string | undefined;
+
+        const log = {
+            level: 'info' as LogLevel,
+            success: false,
+            action: LogActionEnum.AUTH,
+            start: Date.now(),
+            end: Date.now(),
+            timestamp: Date.now(),
+            connection_id: connectionId as string,
+            provider_config_key: providerConfigKey as string,
+            environment_id: environmentId
+        };
+
+        const activityLogId = await createActivityLog(log);
+
+        try {
+            analytics.track(AnalyticsTypes.PRE_APP_STORE_AUTH, accountId);
+
+            if (!providerConfigKey) {
+                errorManager.errRes(res, 'missing_connection');
+
+                return;
+            }
+
+            if (!connectionId) {
+                errorManager.errRes(res, 'missing_connection_id');
+
+                return;
+            }
+
+            const hmacEnabled = await hmacService.isEnabled(environmentId);
+            if (hmacEnabled) {
+                const hmac = req.query['hmac'] as string | undefined;
+                if (!hmac) {
+                    await createActivityLogMessageAndEnd({
+                        level: 'error',
+                        environment_id: environmentId,
+                        activity_log_id: activityLogId as number,
+                        timestamp: Date.now(),
+                        content: 'Missing HMAC in query params'
+                    });
```

Comment:
why adding an activity log message if hmac is missing (or not verified) and not for the other missing params